lib/served/resource/json_api_resource.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+module Served
+  module Resource
+    # Resources that implement JSONApi Spec should use this class
+    class JsonApiResource < Served::Resource::Base
```

Comment:
because you are re-implementing large swaths of code that already exists. If the internal method on how Base changes we now  have to smiliarly refactor major portions of JsonApiResource.

Second, a Resource is a Resource regardless if it is JSON, XML or whatever else. This steps further away from that concept. The original plan/design centered around an end user being able to implement their own module that would know how to serialize/deserialize the content returned by the endpoint, but otherwise would leave the resource class untouched.

Existing example is ActiveRecord. We don't have a PostgresqlBase and a MysqlBase that extend Base instead the serialization is abstracted away so we have a single universal method of handling records. The concept behind served is similar.